src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -168,14 +174,16 @@ def _extract_message(
         try:
             self.schema_validator(use_case_id.value, parsed_payload)
         except ValidationError:
-            if settings.SENTRY_METRICS_INDEXER_RAISE_VALIDATION_ERRORS:
```

Comment:
I am surprised this test is still passing? tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py

also it shouldn't be mutating settings like it is -- probably should be using `with override_settings(...)` either way

Reply:
ah yes, it turns out that `SENTRY_METRICS_INDEXER_RAISE_VALIDATION_ERRORS` is not actually being used in the code paths for test_gen_metrics_multiprocess_steps.py. I can refactor the invalid message test in that file to make this clearer. I can also remove the override settings.

Reply:
the purpose of this was to reject invalid messages in tests 100% of the time, and only warn in prod. we can add a sample rate but i'd force it to 100% in tests again.

Reply:
these are assigning directly to settings -- they should use `override_settings` so it is undone afterwards instead of leaving state for other tests

`git grep 'settings.SENTRY_MET.* ='`

Reply:
> they should use override_settings so it is undone afterwards instead of leaving state for other tests

that's what I am saying, this is not monkeypatching `django.conf` directly, it assigns to the fixture provided by `pytest-django`

i think we had a discussion a few years ago though where you said that using fixtures for monkeypatching comes with suboptimal teardown order

anyway, what i wanted to achieve with this is to have this setting change apply to the entire file, not sure if there's a better way to do it

Reply:
yep:

```python
@pytest.fixture(autouse=True, scope='module')
def apply_setting():
    with override_settings(...):
        yield
```

Reply:
ok, that comes with better-defined teardown order or is the main reason you prefer this because of the larger pytest scope?

Reply:
yeah that will be FIFO -- whereas `monkeypatch`-like fixtures can have out-of-order teardown